third_party/move/move-compiler-v2/src/file_format_generator/function_generator.rs

```diff
@@ -136,15 +137,25 @@ impl<'a> FunctionGenerator<'a> {
                 code: vec![],
             };
             let target = ctx.targets.get_target(&fun_env, &FunctionVariant::Baseline);
-            let code = fun_gen.gen_code(&FunctionContext {
+            let mut code = fun_gen.gen_code(&FunctionContext {
                 module: ctx.clone(),
                 fun: target,
                 loc: loc.clone(),
                 type_parameters: fun_env.get_type_parameters(),
                 def_idx,
             });
-            // Write the spec block table back to the environment.
-            if !fun_gen.spec_blocks.is_empty() {
+            if fun_gen.spec_blocks.is_empty() {
+                // Only consider calling peephole optimizations if there are no spec blocks.
```

Comment:
What are the specific problems with optimizing code with spec blocks?  Is this not a test that can be done in the `peephole_optimizer` code?

Reply:
There are spec block data structures that contain references to code offsets that are no longer valid after potential reordering of code offsets due to peephole optimizations. For now, the simpler path is to not do the optimization if the rarely used "inline" spec blocks.

I have updated the comment to say "inline" spec blocks.